cache/src/test/java/io/envoyproxy/controlplane/cache/SnapshotTest.java

```diff
@@ -109,6 +110,54 @@ public void createSeparateVersionsSetsResourcesCorrectly() {
     assertThat(snapshot.routes().version()).isEqualTo(routesVersion);
   }
 
+  @Test
+  public void createSeparateVersionsForEachCluster() {
+    final String clustersVersion = UUID.randomUUID().toString();
+    final String listenersVersion = UUID.randomUUID().toString();
+    final String routesVersion = UUID.randomUUID().toString();
+    final String secretsVersion = UUID.randomUUID().toString();
+
+    final Map<String, String> endpointVersions = ImmutableMap.of(
+        CLUSTER_NAME, UUID.randomUUID().toString()
+    );
+
+    Snapshot snapshot = Snapshot.create(
+        ImmutableList.of(CLUSTER), r -> clustersVersion,
+        ImmutableList.of(ENDPOINT), resourceNames -> {
```

Comment:
I think we can omit the custom logic here and make this test simpler. The custom logic is already checked in `SnapshotResourcesTest`

Reply:
this test could probably be removed then, what do you think @jakubdyszkiewicz

Reply:
Yeah, I didn't like this test in a first place, but added it because of code coverage.

from contributing:
```
PRs are expected to have 100% test coverage for added code. This can be verified with a coverage build. If your PR cannot have 100% coverage for some reason please clearly explain why when you open it.
```

Reply:
@jakubdyszkiewicz let's see what @snowp thinks

Reply:
yeah im a fan of simplifying the tests if possible

Reply:
can i blow up this entire test then? doesn't seem to be testing anything

Reply:
If it's not testing anything but aids code coverage is it possible to add a useful test for whatever it's covering? If it's not required for code coverage then yeah feel free to remove it

Reply:
i don't think it's really testing anything or helping with coverage, will remove it